libclamav/ole2_extract.c

```diff
@@ -1756,6 +2024,423 @@ ole2_read_header(int fd, ole2_header_t *hdr)
+        cli_warnmsg("ole2: Invalid fExternal flags.  If fExternal bit is set, nothing else can be\n");
+        goto done;
+    }
+
+#define SE_HEADER_FAES (1 << 5)
+    if (SE_HEADER_FAES & headerPtr->flags) {
+        if (!(SE_HEADER_FCRYPTOAPI & headerPtr->flags)) {
+            cli_warnmsg("ole2: Invalid combo of fAES and fCryptoApi flags\n");
+            goto done;
+        }
+
+        cli_dbgmsg(  "Flags: AES\n");
+    }
+
+    cli_dbgmsg(  "Size            = 0x%x\n", headerPtr->size);
+
+    if (headerPtr->flags != headerPtr->encryptionInfo.flags) {
+        cli_warnmsg("ole2: Flags must match\n");
+        goto done;
+    }
+
+    if (0 != headerPtr->encryptionInfo.sizeExtra) {
+        cli_warnmsg("ole2: Size Extra must be 0\n");
+        goto done;
+    }
+
+#define SE_HEADER_EI_AES128 0x0000660e
+#define SE_HEADER_EI_AES192 0x0000660f
+#define SE_HEADER_EI_AES256 0x00006610
+
+#define SE_HEADER_EI_RC4 0x00006801
+
+    switch (headerPtr->encryptionInfo.algorithmID) {
+        case SE_HEADER_EI_AES128:
+            break;
+        case SE_HEADER_EI_AES192:
+            //not implemented
+            goto done;
+        case SE_HEADER_EI_AES256:
+            //not implemented
+            goto done;
+        case SE_HEADER_EI_RC4:
+            //not implemented
+            goto done;
+        default:
+            cli_dbgmsg( "ole2: Invalid Algorithm ID: 0x%x\n", headerPtr->encryptionInfo.algorithmID);
+            goto done;
+    }
+
+#define SE_HEADER_EI_SHA1 0x00008004
+    if (SE_HEADER_EI_SHA1 != headerPtr->encryptionInfo.algorithmIDHash) {
+        cli_warnmsg("ole2: Invalid Algorithm ID Hash: 0x%x\n", headerPtr->encryptionInfo.algorithmIDHash);
+        goto done;
+    }
+
+//#define SE_HEADER_EI_AES128_KEYSIZE 0x00000080
+//#define SE_HEADER_EI_AES192_KEYSIZE 0x000000c0
+//#define SE_HEADER_EI_AES256_KEYSIZE 0x00000100
+
+    switch (headerPtr->encryptionInfo.keySize) {
+        case SE_HEADER_EI_AES128_KEYSIZE:
+            break;
+        case SE_HEADER_EI_AES192_KEYSIZE:
+            break;
+        case SE_HEADER_EI_AES256_KEYSIZE:
+            break;
+        default:
+            cli_warnmsg("ole2: Invalid key size: 0x%x\n", headerPtr->encryptionInfo.keySize);
+            goto done;
+    }
+    cli_dbgmsg( "KeySize = 0x%x\n", headerPtr->encryptionInfo.keySize);
+
+#define SE_HEADER_EI_AES_PROVIDERTYPE 0x00000018
+    if (SE_HEADER_EI_AES_PROVIDERTYPE != headerPtr->encryptionInfo.providerType) {
+        cli_warnmsg("ole2: WARNING: Provider Type should be '0x%x', is '0x%x'\n",
+                    SE_HEADER_EI_AES_PROVIDERTYPE, headerPtr->encryptionInfo.providerType);
+        goto done;
+    }
+
+    cli_dbgmsg( "Reserved1:  0x%x\n", headerPtr->encryptionInfo.reserved1);
+
+    if (0 != headerPtr->encryptionInfo.reserved2) {
+        cli_warnmsg("ole2: Reserved 2 must be zero, is 0x%x\n", headerPtr->encryptionInfo.reserved2);
+        goto done;
+    }
+
+    /*The encryption info is at the end of the CPSName string.  
```

Comment:
Missing space between `/*` and `The`.  In general it would be more readable for spaces after comment characters, but here it looks extra funny because it has them below.